### DISCUSSION TITLE
Why was MongoDB storage engine removed?

### DISCUSSION THREAD

**Discussion by PegasisForever** [umputun/remark42#955] [Q&A] [answered]
Hi, I noticed the MongoDB storage engine was added before and then removed for some reason.

Why is it removed? Is MongoDB being replaced by jrpc? Can I still use MongoDB now?

**Comment by umputun** [comment, marked as answer]
A few reasons:

1. We didn't see much demand for this functionality back in the days. In fact, we saw no real-life usage at all
2. support the alternative/additional storage implementation as a part of the core wasn't such a good idea, so we made a way to connect any external store via jrpc and even provided a simple example for doing it
3. The final reason - I was planning to refactor mongo support with the modern mongo-driver, but instead of spending time on this decision was to drop the support completely for now
4. Another reason - any use case I had in mind for mongo storage required much more, not just storage. Things like external cache, auth-related changes, and so on. We had no time and resources to deal with this as a part of the core and the idea was to allow users to deal with this if they really have such a need.
5. Resurrecting mongo support in some form could be a huge support headache. Currently, the vast majority of support requests are around "how do I configure nginx/docker/etc" Adding a much more complicated mongo piece into this picture (if people start to use it for real) may cause another set of question like "how to configure mongo ... why I can't connect to mongo" and so on

hope it helps

**Comment by ichenhe** [comment]
The last reason is very practical 😆
